frontend/src/utilities/__tests__/router.spec.ts

```diff
@@ -0,0 +1,69 @@
+import { testHook } from '~/__tests__/unit/testUtils/hooks';
+import { setQueryArgument, removeQueryArgument } from '~/utilities/router';
+
+// Mock the NavigateFunction
+const navigate = jest.fn();
+
+describe('setQueryArgument', () => {
+  it('should set a query argument and navigate', () => {
+    // Define the key and value for the query parameter
+    const key = 'param';
+    const value = 'example';
+    // Ensure the query parameter does not exist initially
+    const initialUrl = new URL(window.location.href);
+    initialUrl.searchParams.set(key, value);
+    // Call the hook to set the query parameter
```

Comment:
This is NOT a hook.

Reply:
Thanks, @christianvogt, for pointing this out. I apologize for this oversight.